airflow/api_connexion/endpoints/task_instance_endpoint.py

```diff
@@ -461,13 +436,8 @@ def get_task_instances_batch(session: Session = NEW_SESSION) -> APIResponse:
     )
 
 
-@security.requires_access(
-    [
-        (permissions.ACTION_CAN_EDIT, permissions.RESOURCE_DAG),
-        (permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG_RUN),
-        (permissions.ACTION_CAN_EDIT, permissions.RESOURCE_TASK_INSTANCE),
-    ],
-)
+@security.requires_access_dag("GET", DagAccessEntity.RUN)
```

Comment:
I think this change revealed a bug in the original permissions. Both RUN and TASK_INSTANCE should be PUT (the code actually modifies DagRun state.

Reply:
I removed `@security.requires_access_dag("GET", DagAccessEntity.RUN)`